src/diffusers/loaders/lora_base.py

```diff
@@ -356,21 +356,13 @@ def _load_lora_into_text_encoder(
             text_encoder_lora_state_dict = convert_state_dict_to_diffusers(text_encoder_lora_state_dict)
 
             # convert state dict
-            text_encoder_lora_state_dict = convert_state_dict_to_peft(text_encoder_lora_state_dict)
-
-            for name, _ in text_encoder_attn_modules(text_encoder):
-                for module in ("out_proj", "q_proj", "k_proj", "v_proj"):
-                    rank_key = f"{name}.{module}.lora_B.weight"
-                    if rank_key not in text_encoder_lora_state_dict:
-                        continue
-                    rank[rank_key] = text_encoder_lora_state_dict[rank_key].shape[1]
-
-            for name, _ in text_encoder_mlp_modules(text_encoder):
-                for module in ("fc1", "fc2"):
-                    rank_key = f"{name}.{module}.lora_B.weight"
-                    if rank_key not in text_encoder_lora_state_dict:
-                        continue
-                    rank[rank_key] = text_encoder_lora_state_dict[rank_key].shape[1]
+            text_encoder_lora_state_dict = convert_state_dict_to_peft(text_encoder_lora_state_dict, original_type=StateDictType.DIFFUSERS)
+
+            for name, module in text_encoder.named_modules():
+                if "lora_A" not in name and "lora_B" not in name and isinstance(module, (nn.Linear, nn.Conv2d)):
+                    rank_key = f"{name.removesuffix(".base_layer")}.lora_B.weight"
+                    if rank_key in text_encoder_lora_state_dict:
+                        rank[rank_key] = text_encoder_lora_state_dict[rank_key].shape[1]
```

Comment:
These changes support TE LoRA other than CLIP; but they cause one other model to fail, likely an edge case.

Reply:
I think it's fine to not add these changes in this PR as it seems unrelated.

Reply:
I'll make a separate PR for this and we can review that one first -- it seems to be the crux of what's failing with the tests IIUC. Then coming back to this one should be easy.

Reply:
It still causes the failures here, and I'm kinda clueless how to fix them: https://github.com/huggingface/diffusers/pull/10216#issuecomment-2577850739

Reply:
I don't think so.

https://github.com/huggingface/diffusers/blob/edb8c1bce67e81f0de90a7e4c16b2f6537d39f2d/src/diffusers/loaders/lora_pipeline.py#L2560 doesn't have text encoder either.

Reply:
neither does OmniGen